docs/book/src/tasks/experimental-features/runtime-sdk/implement-topology-mutation-hook.md

```diff
@@ -71,6 +167,14 @@ This section outlines considerations specific to Topology Mutation hooks:
   always return the same error message. Otherwise the system might became unstable due to controllers being overloaded
   by continuous changes to Kubernetes resources as these messages are reported as conditions. See [error messages](implement-extensions.md#error-messages).
 
+### Variable discovery guidelines
+* **Unique variable names**: When using external variable discovery, care should be taken to ensure that variable names are unique.
+  If the names are not unique the ClusterClass controller will attempt to deduplicate variables by comparing their definition. If this
+  is not possible the variables will be namespaced and must be referenced by their namespace when defined in a Cluster.
+* **Avoid updating variable definitions**: When using external variables updating the definitions of variables in use by existing Clusters
+  can cause reconciliation to fail where existing variable values are no longer valid, or where changing the definitions results in namespacing of
+  variables which were previously not namespaced.
```

Comment:
What about
```suggestion
* **Avoid breaking changes variable definitions**: Changing a variable definition can lead to problems on existing clusters because reconciliation will stop if variable values will not match anymore the new definition.
Furthermore, when the variables defined for this patch are going to be used in combinations with other patches providing variables with the same name, changes to variable definition can lead to the system requiring the users to provide different values for each namespace the value exists in.
As a consequence updates to the variable definition should always be carefully evaluated, and very well documented in the extension release notes, so ClusterClass authors can evaluate impacts of changes before upgrade.
```